PenLineModel.h

```diff
@@ -80,11 +92,15 @@ class CAVEWHERE_SKETCH_LIB_EXPORT PenLineModel : public QAbstractItemModel {
 
 signals:
     void currentStrokeWidthChanged();
-
+    void undoStackChanged();
 
 private:
+    friend class PenLineModelUndoCommand;
+
     Q_OBJECT_BINDABLE_PROPERTY_WITH_ARGS(PenLineModel, double, m_currentStrokeWidth, 2.5, &PenLineModel::currentStrokeWidthChanged);
     QVector<PenLine> m_lines;
+    QUndoStack m_undoStack;
+    PenLineModelUndoCommand *m_activeUndoCommand;
```

Comment:
This shouldn't be stored. Once pushed the qundostack should take ownership of it. https://doc.qt.io/qt-6/qundostack.html#push